### PR TITLE
WIP 1.2.0

### PR DESCRIPTION
**Pull request by KyrneDev**
- [x] - Poll close date, #18 
- [x] - Public/Private polls, #23
- [x] - Change poll answer, #13

### CODE REVIEW

**Comment by KyrneDev** [comment]
PR is finished, awaiting code review

**Review by dsevillamartin — CHANGES_REQUESTED** [review]
- Button to see voters moves after voting.
image
image

- If possible, there should be an option for polls to never expire
- If no voters, modal should say "none"
- Voters' avatars should be properly sized
image
- Admin: what does `Allow poll editing` mean? Clarify
    - Any member can edit the poll?
    - Voter can change their poll vote?

**Comment by KyrneDev** [comment]
So if you leave the end date blank it will never expire

I'll fix the style

That admin perm is part of 1.1.0, not this version, it allows admins to edit polls

**Comment by dsevillamartin** [comment]
@CDK2020 There's two permissions, one in "participate" and one in "moderate".

image
image

**Comment by KyrneDev** [comment]
The first screenshot doesn't contain the edit poll permission. I assume you are talking about "self edit polls" which allows the user to edit his/her own poll. The mod item allows mods to edit polls

**Comment by dsevillamartin** [comment]
@CDK2020 Meant to send this. I know it means that you can edit your own poll, but some users may be confused as to why there are two permissions for seemingly the same thing.

image
vs
image

**Comment by ghost** [comment]
Maybe change it to “Allow users to edit their polls”?

**Comment by KyrneDev** [comment]
Yup, it was meant to say that, thanks for pointing it out. I'll fix everything tonight. I'll also update the end date placeholder to say "optional"

**Comment by KyrneDev** [comment]
@datitisev Updated, ready for re-review

**Comment by dsevillamartin** [comment]
@CDK2020 What about polls that don't expire? It doesn't seem to be working, I leave expiration date empty and it says the poll has ended and doesn't let me vote or edit it.

**Comment by tjrgg** [comment]
image
I can confirm @datitisev's issue. It appears to happen for me as well.

Edit: This is now fixed.

**Review by tjrgg — COMMENTED** [review]
There are a couple of things I've noticed that I'd personally like to see changed.

First off though, a bug I seemed to have discovered was that after I started a discussion with a poll, I clicked on the tag next to the title of the discussion at the top of the page to return to the discussion list. I then went to start another discussion and noticed that the button that normally says "Add Poll" now said "Edit Poll". When I click on what now says "Edit Poll", I'm presented with the "Add Poll" screen. I can only guess this has something to do with the fact that the page was not refreshed or reloaded at all after the initial poll was created.

Barring that, here are few things I noticed during my first hour of using the extension.
- There is some inconsistency in the "Add Poll" modal. "Question 1" and "Answer 1" all appear in this dark gray color, but the "Poll end date" at the bottom appears in white. It looks a bit off, in my opinion.
image
- The "Edit Poll" option doesn't have a "Save" or "Submit" button, which can be quite confusing for the user, as it was for me at first. I later discovered that closing the modal refreshed my page with my changes, although that's not what I would expect to experience within Flarum. Everything in Flarum is smooth and updates without reloading the entire page. In addition, the lack of a "Save" or "Submit" button can confuse the user. So, I'd personally like to see at least a button, even if it just closes the modal and refreshes the page, but ideally, it just updates the poll without a refresh, similar to how posts get updated.
image
- The "Remove Poll" option uses a standard JavaScript popup. For consistency sake, I'd love it to use one of those Flarum modal popups for confirmation instead. 
image
- The "Public poll" button is a bit wordy. I would prefer a short and concise button. "View Voters" would do just fine, in my opinion. Ideally, that would be hidden under the "Reply" button menu or at the bottom of the post where the "Like" and "Reply" links are, but either way, it needs less words.
image

**Comment by tjrgg** [comment]
I also have to suggest splitting up the edit and delete permissions for polls. The use case I mentioned in flarum/core#1387 applies here as well, so I have to make the same suggestion here.

**Comment by Ralkage** [comment]
@Tyler0130 brought up similar points as yourself, these issues will definitely be fixed soon, especially with the missing "Save" button and wordy "Public poll: Click to view voters button" (There is an issue where the yml is having problems parsing this string due to the colon ":" afer "poll").
